core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -31,196 +32,50 @@
 import javax.annotation.Nullable;
 
 /**
- * Abstract file source config
+ * File source config
  */
-public abstract class FileSourceConfig extends ReferencePluginConfig {
-  protected static final String MAX_SPLIT_SIZE_DESCRIPTION = "Maximum split-size for each mapper in the MapReduce " +
-    "Job. Defaults to 128MB.";
-  protected static final String TABLE_DESCRIPTION = "Name of the Table that keeps track of the last time files " +
-    "were read in. If this is null or empty, the Regex is used to filter filenames.";
-  protected static final String INPUT_FORMAT_CLASS_DESCRIPTION = "Name of the input format class, which must be a " +
-    "subclass of FileInputFormat. Defaults to a CombinePathTrackingInputFormat, which is a customized version of " +
-    "CombineTextInputFormat that records the file path each record was read from.";
-  protected static final String REGEX_DESCRIPTION = "Regex to filter out files in the path. It accepts regular " +
-    "expression which is applied to the complete path and returns the list of files that match the specified pattern." +
-    "To use the TimeFilter, input \"timefilter\". The TimeFilter assumes that it " +
-    "is reading in files with the File log naming convention of 'YYYY-MM-DD-HH-mm-SS-Tag'. The TimeFilter " +
-    "reads in files from the previous hour if the field 'timeTable' is left blank. If it's currently " +
-    "2015-06-16-15 (June 16th 2015, 3pm), it will read in files that contain '2015-06-16-14' in the filename. " +
-    "If the field 'timeTable' is present, then it will read in files that have not yet been read. Defaults to '.*', " +
-    "which indicates that no files will be filtered.";
-  protected static final String FILESYSTEM_PROPERTIES_DESCRIPTION = "A JSON string representing a map of properties " +
-    "needed for the distributed file system.";
-  protected static final String FORMAT_DESCRIPTION = "Format of the file. Must be 'text', 'avro' or " +
-    "'parquet'. Defaults to 'text'.";
+public class FileSourceConfig extends AbstractFileSourceConfig {
 
   private static final Gson GSON = new Gson();
   private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
-  @VisibleForTesting
-  static final long DEFAULT_MAX_SPLIT_SIZE = 134217728;
 
-  @Nullable
-  @Description(FILESYSTEM_PROPERTIES_DESCRIPTION)
-  @Macro
-  public String fileSystemProperties;
-
-  @Nullable
-  @Description(REGEX_DESCRIPTION)
-  @Macro
-  public String fileRegex;
-
-  @Nullable
-  @Description(TABLE_DESCRIPTION)
-  @Macro
-  public String timeTable;
-
-  @Nullable
-  @Description(INPUT_FORMAT_CLASS_DESCRIPTION)
   @Macro
-  public String inputFormatClass;
```

Comment:
if they tried to do that it would have failed anyway. The class exposed this as a property, but had no way of converting whatever key/value the format provided into a StructuredRecord. So it was never working.